### PR TITLE
Validation: Mandatory handling unit

### PR DESCRIPTION
**Pull request by viralkansodiya**
if stock entry type is not 'Material Receipt'  and Item has enabled handling unit feature then handling unit should be mandatory

### CODE REVIEW

**Comment by viralkansodiya** [comment]
Process to merge this PR after merging #88

File: beam/beam/overrides/stock_entry.py
Comment (agritheory) [inline beam/beam/overrides/stock_entry.py]: You can use `row.idx` instead and remove `enumerate`.

**Review by agritheory — CHANGES_REQUESTED** [review]

File: beam/beam/overrides/stock_entry.py
Comment (agritheory) [inline beam/beam/overrides/stock_entry.py]: To follow Frappe conventions, use 'doc' to refer to the document object when it is not a class method and `self` when you're writing a class method (including overrides).

File: beam/beam/overrides/stock_entry.py
Comment (agritheory) [inline beam/beam/overrides/stock_entry.py]: Please refactor to f-string and add `frappe._(...)` translation helper

**Review by agritheory — CHANGES_REQUESTED** [review]
Please also rebase

File: beam/beam/overrides/stock_entry.py
Comment (viralkansodiya) [reply beam/beam/overrides/stock_entry.py]: Changes has been done as you mentioned.

File: beam/beam/overrides/stock_entry.py
Comment (viralkansodiya) [reply beam/beam/overrides/stock_entry.py]: Changes has been done as you mentioned

**Review by agritheory — APPROVED** [review]

**Comment by agritheory** [comment]
@viralkansodiya With this feature added, some tests are now failing and we should address that. 

To align with the bakery narrative of the tests, "Water" and "Ice Water" items should not have handling units and we should change this in `tests/setup.py`. 

To fix the repack tests, we only need to add the handling unit key collected [here](https://github.com/agritheory/beam/actions/runs/8627238494/job/23646825549?pr=92#step:11:29) into [this dictionary](https://github.com/agritheory/beam/actions/runs/8627238494/job/23646825549?pr=92#step:11:41). That said, the quality of this test that I originally wrote could be improved by simulating a scan, like we do [here](https://github.com/agritheory/beam/blob/version-14/beam/tests/test_scan.py#L16). 

Let me know if you run into any trouble.

**Comment by viralkansodiya** [comment]
@agritheory 
At this point, we need to modify the validation of mandatory handling units based on the type of stock entry.

For example:

1. If the stock entry type is 'Repack', the handling unit should be mandatory unless it is a finished item.

2. If the stock entry type is 'Manufacturing', the handling unit should be mandatory only for raw materials, not for finished items.

Similarly, we should consider other types of stock entry as well."

Please give your suggestion

**Pull request merged**
